static/app/components/teamSelector.tsx

```diff
@@ -357,11 +359,28 @@ function TeamSelector(props: Props) {
     [includeUnassigned, multiple, stylesProp]
   );
 
+  useEffect(() => {
+    // Only take action after we've finished loading the teams
+    if (fetching) {
+      return;
+    }
+
+    // If there is only one team, and our flow wants to enable using that team as a default, update the parent state
+    if (options.length === 1 && defaultUseIfOnlyOne) {
+      const castedValue = multiple
+        ? (options as TeamOption[])
+        : (options[0] as TeamOption);
+      handleChange(castedValue);
+    }
+    // We only want to do this once when the component is finished loading for teams and mounted
+    // If the user decides they do not want the default, we should not add the default value back
+  }, [fetching, defaultUseIfOnlyOne]);
```

Comment:
I'm not sure I understand the logic change here. What exactly was the bug?

Reply:
This is not related to the bug, this is part of the improvements. We want to default using the team if there is only 1 team available from the options

Reply:
I see. Maybe we can split this out into it's own PR in that case?

Reply:
sure!